net/dns/resolver/tsdns_test.go

```diff
@@ -1417,3 +1417,41 @@ func TestUnARPA(t *testing.T) {
 		}
 	}
 }
+
+func TestServfail(t *testing.T) {
```

Comment:
Can you add a little comment above this test saying what the overall goal of the test is, what it's testing? Maybe a bug link if applicable?